include/revng/AutoEnforcer/AutoEnforcer.h

```diff
@@ -0,0 +1,183 @@
+                      Registry.createEmpty(),
+                      std::forward<EnforcerWrappers>(EnfWrappers)...));
+  }
+
+  llvm::Error deduceInvalidations(InvalidationMap &Invalidations) const {
+    return Pipeline.getInvalidations(Invalidations);
+  }
+
+  const BackingContainers &getStartingContainers() const {
+    return Pipeline.getStartingContainers();
+  }
+
+  BackingContainers &getStartingContainers() {
+    return Pipeline.getStartingContainers();
+  }
+
+  template<typename BackingContainerType>
+  const BackingContainerType &getStartingContainer(llvm::StringRef Name) const {
+    return Pipeline.getStartingContainer<BackingContainerType>(Name);
+  }
+
+  template<typename BackingContainerType>
+  BackingContainerType &getOrCreateStartingContainer(llvm::StringRef Name) {
+    return Pipeline.getOrCreateStartingContainer<BackingContainerType>(Name);
+  }
+
+  bool isStartingContainerEmpty(llvm::StringRef Name) const {
+    return Pipeline.getStartingContainers().isContainerEmpty(Name);
+  }
+
+  const BackingContainers &getFinalContainers() const {
+    return Pipeline.getFinalContainers();
+  }
+  BackingContainers &getFinalContainers() {
+    return Pipeline.getFinalContainers();
+  }
+
+  template<typename BackingContainerType>
+  const BackingContainerType &getFinalContainer(llvm::StringRef Name) const {
+    return Pipeline.getFinalContainer<BackingContainerType>(Name);
+  }
+
+  template<typename BackingContainerType>
+  BackingContainerType &getOrCreateFinalContainer(llvm::StringRef Name) {
+    return Pipeline.getOrCreateFinalContainer<BackingContainerType>(Name);
+  }
+
+  llvm::Error
+  run(const BackingContainersStatus &Targets, llvm::raw_ostream *OS = nullptr) {
+    return Pipeline.run(*AEContext, Targets, OS);
+  }
+
+  template<typename BackingContainerType, typename... Args>
+  void registerContainerFactory(llvm::StringRef Name, Args &&...Arguments) {
+    revng_assert(not CommittedRegistry);
+    using BCT = BackingContainerType;
+    Registry.registerContainerFactory<BCT>(Name,
+                                           std::forward<Args>(Arguments)...);
+  }
+
+  void
+  registerContainerFactory(llvm::StringRef Name,
+                           std::unique_ptr<BackingContainerFactory> Entry) {
+    Registry.registerContainerFactory(Name, std::move(Entry));
+  }
+
+  template<typename BackingContainer>
+  void registerDefaultConstructibleFactory(llvm::StringRef Name) {
+    revng_assert(not CommittedRegistry);
+    Registry.registerDefaultConstructibleFactory<BackingContainer>(Name);
+  }
+
+  iterator begin() { return Pipeline.begin(); }
+  iterator end() { return Pipeline.end(); }
+
+  const_iterator begin() const { return Pipeline.begin(); }
+  const_iterator end() const { return Pipeline.end(); }
+
+  template<typename OStream>
+  void dump(OStream &OS, size_t Indents = 0) const {
+    Pipeline.dump(OS, Indents);
+  }
+  void dump() const debug_function { dump(dbg); }
```

Comment:
Move this down and delimit it with a `public:`.
Let's try to group related things together.

Reply:
ok i guess i will do all this kinds of things in block and move around all fields of  classes in headers in one go

Reply:
OK, if you want we can have a special comment for this. Like `[FINALCLEANUPS]` and then leave the discussion open.